# Breast Cancer Detection — Logistic Regression

Binary classification model predicting malignant vs benign tumors using the Wisconsin Breast Cancer dataset (scikit-learn built-in).

In [45]:
from sklearn.datasets import load_breast_cancer
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import RandomForestClassifier,GradientBoostingClassifier
from sklearn.svm import SVC
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import accuracy_score,precision_score,recall_score,f1_score,confusion_matrix
import pandas as pd
import numpy as np
import joblib

bc=load_breast_cancer()
df=pd.DataFrame(bc.data,columns=bc.feature_names)
df["target"] = bc.target
X=df.drop("target",axis=1)
y=df['target']
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.2,random_state=42)
scaler=StandardScaler()
X_train_scaled=scaler.fit_transform(X_train)
X_test_scaled=scaler.transform(X_test)
LR=LogisticRegression()
LR.fit(X_train_scaled,y_train)
y_pred=LR.predict(X_test_scaled)
print("Accuracy_score:",accuracy_score(y_test,y_pred))
print("Precision_score:",precision_score(y_test,y_pred))
print("recall_score:",recall_score(y_test,y_pred))
print("f1_score:",f1_score(y_test,y_pred))
print("Confusion_matrix:\n",confusion_matrix(y_test,y_pred))


Accuracy_score: 0.9736842105263158
Precision_score: 0.9722222222222222
recall_score: 0.9859154929577465
f1_score: 0.9790209790209791
Confusion_matrix:
 [[41  2]
 [ 1 70]]


In [41]:
joblib.dump(LR,"Breast_cancer_detection.joblib")

['Breast_cancer_detection.joblib']

In [42]:
model = joblib.load("Breast_cancer_detection.joblib")

In [43]:
y_pred1 = model.predict(X_test_scaled)

In [44]:
print("Accuracy_score:",accuracy_score(y_test,y_pred))
print("Precision_score:",precision_score(y_test,y_pred))
print("recall_score:",recall_score(y_test,y_pred))
print("f1_score:",f1_score(y_test,y_pred))
print("Confusion_matrix:\n",confusion_matrix(y_test,y_pred))

Accuracy_score: 0.9736842105263158
Precision_score: 0.9722222222222222
recall_score: 0.9859154929577465
f1_score: 0.9790209790209791
Confusion_matrix:
 [[41  2]
 [ 1 70]]


## Summary

**Dataset:** scikit-learn's built-in Wisconsin Breast Cancer dataset (569 samples, 30 features)

**Steps:** Split data 80/20 → scaled features with StandardScaler → trained Logistic Regression → evaluated on test set → saved model with joblib and verified reload gives identical predictions.

**Results:** Accuracy 97.4%, Precision 97.2%, Recall 98.6%, F1 97.9%.

**Next steps:** Compare against Random Forest/SVM/Gradient Boosting, add feature importance analysis.